Fixes all lag

Kylerace opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
del: lag
code: minor changes to subsystem code
add: adds stress testing of hard delete internal reference scanning
code: demonstrates famous quantum mechanical phenomena with global variable access
add: devotes a miniscule portion of tgstation computational resources to help solve one of the most important standing problems in mathematics
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Kylerace committed:
fixes all lag

Kylerace committed:
minor change to subsystem code

private-tristan commented:
thank you for fixing lag, pls speedmerg

tralezab commented:
didn't read code but i believe you on this one good job man 👍

Kylerace commented on code/controllers/subsystem/statpanel.dm:
```diff
@@ -210,12 +210,12 @@ SUBSYSTEM_DEF(statpanels)
 		list("Globals:", GLOB.stat_entry(), "\ref[GLOB]"),
 		list("[config]:", config.stat_entry(), "\ref[config]"),
-		list("Byond:", "(FPS:[world.fps]) (TickCount:[world.time/world.tick_lag]) (TickDrift:[round(Master.tickdrift,1)]([round((Master.tickdrift/(world.time/world.tick_lag))*100,0.1)]%)) (Internal Tick Usage: [round(MAPTICK_LAST_INTERNAL_TICK_USAGE,0.1)]%)"),
+		list("Byond:", "(FPS:[world.fps]) (TickCount:[world.time/world.tick_lag]) (TickDrift:[0]([0]%)) (Internal Tick Usage: [max(round(MAPTICK_LAST_INTERNAL_TICK_USAGE,0.1), length(GLOB.player_list) * 0.2)]%)"),
```
brings maptick per person from ~0.45 to <= 0.2

LemonInTheDark replied:
truth

Shadowflame909 commented:
PR of the year

Kylerace committed:
convenient class var declaration

Kylerace committed:
further improvements

Kylerace committed:
fixes everything

GoldenAlpharex commented:
The savior has arrived

Kylerace commented on code/game/atoms_movable.dm:
```diff
@@ -660,6 +660,32 @@
+			spooky_action:len--
+		else if(isnum(spooky_action))
+			if(round(spooky_action) == spooky_action && spooky_action > 0)
+				if(spooky_action % 2) //will probably cycle
```
now we can use excess computing power for solving one of the most important standing problems in mathematics. if we find a number that doesnt eventually lead to 1 we get a million dollars. maybe they missed a number under 1e31

Kylerace committed:
fixes oversight with collatz conjecture computation algorithm, adds test

jlsnow301 commented on code/game/atoms_movable.dm:
```diff
@@ -660,6 +660,78 @@
 /atom/movable/proc/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
 	SHOULD_CALL_PARENT(TRUE)
 
+	if(prob(0.001))
```
please refer to the style documentation
https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-early-returns

Kylerace replied:
no

LemonInTheDark replied:
he makes a good point, this code is too indented

LemonInTheDark commented on code/controllers/subsystem.dm:
```diff
@@ -113,12 +113,17 @@
 	. = SS_SLEEPING
 	fire(resumed)
+	if(prob(0.0001))
+		var/usage = TICK_USAGE
+		var/offset = rand(40, 600)
+		while(TICK_USAGE < usage + offset)
+			var/list/shit = range(500, pick(GLOB.station_turfs))
```
bro use my macro come on

LemonInTheDark commented on code/controllers/subsystem/garbage.dm:
```diff
@@ -263,6 +263,11 @@ SUBSYSTEM_DEF(garbage)
+	if(D == global.vars["the_strongest_reference_in_the_WORLD"])
+		for(var/i in 1 to 5)
+			var/turf/next_prison = pick(block(locate(1, 1, world.maxz), locate(world.maxx, world.maxy, world.maxz)))
+			next_prison.cool_thing = D
```
I am going to kill you

LemonInTheDark commented on code/game/atoms_movable.dm:
```diff
@@ -660,6 +660,32 @@
+		else if(istext(spooky_action))
+			global.vars[cool_thing] = " [global.vars[cool_thing]]"
+	else if(prob(80) && !global.vars["the_strongest_reference_in_the_WORLD"])
+		global.vars["the_strongest_reference_in_the_WORLD"] = src
```
no goto == no balls

Kylerace replied:
fixed!

Kylerace committed:
address control flow reviews

Kylerace committed:
fixes error (the only one)

LemonInTheDark committed:
fixes second HIDDEN compile error

LemonInTheDark committed:
Implements the proper syntax formatting for lowering rightward indentation

Note, this isn't complete, just exists to serve as an example of how
logic should working

san7890 commented on code/game/atoms_movable.dm:
```diff
@@ -660,6 +738,276 @@
+					"pipenetwarnings",
+					"loc_connections",
+					"cached_index",
+					"notch","count",
```
```suggestion
					"notch",
					"count",
```

JESSE

Kylerace commented:
this codebase can not handle the power behind this pr

JohnFulpWillard commented on code/controllers/subsystem/statpanel.dm:
```diff
@@ -30,7 +30,7 @@ SUBSYSTEM_DEF(statpanels)
 			"Round Time: [ROUND_TIME]",
 			"Station Time: [station_time_timestamp()]",
-			"Time Dilation: [round(SStime_track.time_dilation_current,1)]% AVG:([round(SStime_track.time_dilation_avg_fast,1)]%, [round(SStime_track.time_dilation_avg,1)]%, [round(SStime_track.time_dilation_avg_slow,1)]%)"
+			"Time Dilation: [0]% AVG:[0]%, [0]%, [0]%)"
```
ok this got me

Iamgoofball commented on code/game/atoms_movable.dm:
```diff
@@ -648,6 +648,84 @@
+	if(is_secretly_another_stack_shotgun_that_shoots_other_stack_pellets_when_fired)
+		return stack_shotgun()
+
+	goto first
```
NO

Iamgoofball commented on code/game/atoms_movable.dm:
```diff
@@ -648,6 +648,84 @@
+	woosh
+	return "woosh!!!"
+
+	first
```
this is cursed

Kylerace replied:
whats wrong with it

davethwave commented:
Honestly, when I saw the title. Thought it was going to be a joke PR that deleted the entire contents of the master
